R/utilities.R

```diff
@@ -509,6 +509,191 @@ ExpMean <- function(x) {
   return(log(x = mean(x = exp(x = x) - 1) + 1))
 }
 
+#' Export Seurat object for UCSC cell browser
+#'
+#' @param object Seurat object
+#' @param dir output directory path
+#' @param dataset.name name of the dataset
+#' @param vars vector of metadata fields to export
+#' @param vars.names list that defines metadata field names after the export.
+#' Should map metadata column name to export name
+#' @param reductions vector of reduction names to export
+#' @param markers.file path to file with marker genes
+#' @param cluster.field name of the metadata field containing cell cluster
+#' @param cb.dir in which dir to create UCSC cell browser content root
+#' @param port \emph{experimental} on which port to run UCSC cell browser after export
+#'
+#' @return ...
+#'
+#' @author Maximilian Haeussler, Nikolay Markov
+#'
+#' @importFrom utils browseURL
+#' @importFrom reticulate py_module_available import
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' ExportToCellbrowser(object = pbmc_small, dataset.name = "PBMC", dir = "out")
+#' }
+#'
+ExportToCellbrowser <- function(
+  object,
+  dir,
+  dataset.name,
+  vars = NULL,
+  vars.names = NULL,
+  reductions = "tsne",
+  markers.file = NULL,
+  cluster.field = NULL,
+  port = NULL,
+  cb.dir = NULL
```

Comment:
I'm not sure why this needs to be separate from `dir`, but I don't know cell browser so it could very well need to be different. Could you ensure that the docs are clear as to why these need to be different?